front/lib/actions/configuration/mcp.ts

```diff
@@ -40,59 +42,46 @@ export async function fetchMCPServerActionConfigurations(
     return new Map();
   }
 
-  // Find the associated data sources configurations.
-  const allDataSourceConfigurations =
-    await AgentDataSourceConfiguration.findAll({
-      where: {
-        mcpServerConfigurationId: {
-          [Op.in]: mcpServerConfigurations.map((r) => r.id),
-        },
-      },
+  const whereClause: WhereOptions = {
```

Comment:
"WhereOptions" is not really checking that you're passing the right type of "WhereOptions" no ?